src/main/scala/com/databricks/spark/xml/parsers/CustomTextOutputFormat.java

```diff
@@ -0,0 +1,114 @@
+package com.databricks.spark.xml.parsers;
+
+import org.apache.hadoop.classification.InterfaceAudience;
+import org.apache.hadoop.classification.InterfaceStability;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.compress.CompressionCodec;
+import org.apache.hadoop.io.compress.GzipCodec;
+import org.apache.hadoop.mapred.*;
+import org.apache.hadoop.util.Progressable;
+import org.apache.hadoop.util.ReflectionUtils;
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+
+/**
+ * author: ahuoo
+ * Don't write value in new line, just append
+ */
+@InterfaceAudience.Public
+@InterfaceStability.Stable
+public class CustomTextOutputFormat<K, V> extends FileOutputFormat<K, V> {
```

Comment:
I didn't quite understand of this. We won't be able to read it back if eveything is written without newlines.

Reply:
Maybe I  need to add a switcher in it,  then you will find the difference,  actually, the  IndentingXMLStreamWriter writer already has beautiful format,  we don't need add some new separator in output files.  The default writer will add \n in the xml file

Reply:
Yea, we can avoid indenting XML writer but just use normal compact xml writer. Is the purpose of doing this to save newline characters?

Reply:
Indenting XML writer is good.  Maybe you can make the writer configurable.

Reply:
I mean, how about using just normal writer instead of indenting writer? The change will be minimised and we could reach the same goal.

Reply:
Here is the output format for the same xml. 
Use Default TextOutputFormat
```
<?xml version="2.0" encoding="UTF-8"?>
<ROWS>
    <ROW name="ahuoo" url="www.ahuoo.com"
/>
    <ROW name="google" url="www.google.com"
/>
    <ROW name="facebook" url="++"
/>
</ROWS>
```

Use CustomTextOutputFormat,  Is the the normal writer able to generate this format?
```
<?xml version="2.0" encoding="UTF-8"?>
<ROWS>
    <ROW name="ahuoo" url="www.ahuoo.com"/>
    <ROW name="google" url="www.google.com"/>
    <ROW name="facebook" url="++"/>
</ROWS>

```

Reply:
Can you post this in PR description? I thought you mean we should get rid of indentation. So, is it only useful when to write out selr-closing elements? In that case, let's don't add it.

Reply:
Yes, I can post a new PR
No,  it is also useful for non-self-closing element.

Use Default TextOutputFormat 
<?xml version="2.0" encoding="UTF-8"?>
<ROWS>
    <ROW name="beatles">
        <hit_songs>help</hit_songs>
        <hit_songs>hey jude</hit_songs>
        <selfcloseTag name="beatles"/>
        <some_array>1</some_array>
        <some_array>2</some_array>
        <some_array>3</some_array>
        <some_struct 1="foo" 2="1" 3="0.3"/>
        <some_struct2 x="foo" y="1">0.3</some_struct2>
        <some_map key1="1" key2="2"></some_map>
    </ROW>
    <ROW name="romeo">
        <hit_songs>eres mia</hit_songs>
        <selfcloseTag name="romeo"/>
        <some_array>1</some_array>
        <some_array>2</some_array>
        <some_array>3</some_array>
        <some_struct 1="foo" 2="1" 3="0.3"/>
        <some_struct2 x="foo" y="1">0.3</some_struct2>
        <some_map key1="1" key2="2"></some_map>
    </ROW>
    <ROW name="romeo">
        <hit_songs>++</hit_songs>
        <selfcloseTag name="romeo"/>
        <some_array>1</some_array>
        <some_array>2</some_array>
        <some_array>3</some_array>
        <some_struct 1="foo" 2="1" 3="0.3"/>
        <some_struct2 x="foo" y="1">0.3</some_struct2>
        <some_map key1="1" key2="2"></some_map>
    </ROW>
</ROWS>

Use CustomTextOutputFormat, there are a lot of blank lines
```
<?xml version="2.0" encoding="UTF-8"?>
<ROWS>
    <ROW name="beatles">
        <hit_songs>help</hit_songs>
        <hit_songs>hey jude</hit_songs>
        <selfcloseTag name="beatles"/>
        <some_array>1</some_array>
        <some_array>2</some_array>
        <some_array>3</some_array>
        <some_struct 1="foo" 2="1" 3="0.3"/>
        <some_struct2 x="foo" y="1">0.3</some_struct2>
        <some_map key1="1" key2="2"></some_map>
    </ROW>

    <ROW name="romeo">
        <hit_songs>eres mia</hit_songs>
        <selfcloseTag name="romeo"/>
        <some_array>1</some_array>
        <some_array>2</some_array>
        <some_array>3</some_array>
        <some_struct 1="foo" 2="1" 3="0.3"/>
        <some_struct2 x="foo" y="1">0.3</some_struct2>
        <some_map key1="1" key2="2"></some_map>
    </ROW>

    <ROW name="romeo">
        <hit_songs>++</hit_songs>
        <selfcloseTag name="romeo"/>
        <some_array>1</some_array>
        <some_array>2</some_array>
        <some_array>3</some_array>
        <some_struct 1="foo" 2="1" 3="0.3"/>
        <some_struct2 x="foo" y="1">0.3</some_struct2>
        <some_map key1="1" key2="2"></some_map>
    </ROW>

</ROWS>

```

Reply:
Ah, now I got what you mean. Can you just strip new lines for each xml record at `XmlFiles.saveAsXmlFile`?